### DISCUSSION TITLE
Datasmith animation to fast

### DISCUSSION THREAD

**Discussion by hinchmonster** [iTwin/itwinjs-core#4599] [Q&A] [answered]
When I export the imodel from the datasmith exporter the animation is way to quick so it doesnt sample the growth sequences enough so they aren't smooth they are jumpy. Is there anyway to sample the animation more frequently?

**Comment by arnab-git-hub** [comment]
@MichaelBelousov Can you please help with this issue?

**Comment by MichaelBelousov** [comment]
Hi @hinchmonster the animation is not actually sampled technically. Could you share a gif or screen recording to demonstrate how jumpy it is? This may be a bug at which point you could [make an issue](https://github.com/iTwin/itwinjs-core/issues/new/choose)

**Comment by hinchmonster** [comment]
@MichaelBelousov thanks for getting back to me, I have attached both itwin animation and how it shows in unreal

It especially shows on the lift core how it appears half way up and that none of the steel grows it just appears.

https://user-images.githubusercontent.com/80821830/200342809-4d8605de-c165-4895-a872-600bcc72eb1a.mp4


https://user-images.githubusercontent.com/80821830/200342819-19c18a38-8ba5-4720-9054-d8554078050f.mp4

**Comment by MichaelBelousov** [comment, marked as answer]
Yes this would be a bug. Would you be able to [make an issue](https://github.com/iTwin/itwinjs-core/issues/new/choose) and I'll update you there with progress? I will see if any of our test data exhibits something similar enough for me to debug without seeking our more data.

**Reply by hinchmonster** [reply]
@MichaelBelousov thanks i have just submitted an issue now in the meantime is there an older version where the bug doesn't exist?

**Reply by MichaelBelousov** [reply]
Well I believe yes, but it is not simple at the moment to consume old versions.

Also, the bug was introduced when making the existing solution robust to moving the datasmith scene around in Unreal engine. If you do not move it from where datasmith imports it, then the old solution should work for you. Also other features and fixes may be missing in the old version.

Here are the instructions to do it for now, we intend to eventually make it easier to use an older version:
1. edit the following url to the version you need. In this case, you should try `0.7.1` from before that fix was made: `https://autoupdatecdn.bentley.com/itwinexporterdatasmith/iTwinDatasmithExporter-0.7.1-full.nupkg`
2. rename the downloaded file to `iTwinDatasmithExporter-0.7.1-full.nupkg.zip` (you must change the file extension, `.nupkg` is in fact just a `.zip` file)
3. extract the contents of the new zip file.
4. locate the install directory of iTwin Exporter for Datasmith, it should be in `C:\Users\YOUR_USER_NAME\AppData\Local\iTwinDatasmithExporter\app-0.7.3`, except replace `YOUR_USER_NAME` with your actual user name on your PC
5. copy and paste the contents of the `lib/net45` directory in the newly extracted zip file contents over into your install directory of the iTwin Exporter for Datasmith, making sure to select the options to overwrite any files of the same name.

Unfortunately, the app will still try to update automatically to the old version. If it completes this (and it will prompt you), then restarting will update the old version you just forced. So for now you just have to not close it while using an old version.

**Reply by hinchmonster** [reply]
thanks that version works much better for the animation but i had to unistall itwin and then just run direct from the net45 folder for my pc to recognise it as 0.7.1

**Reply by MichaelBelousov** [reply]
Currently the workaround is to use the 0.7.1 datasmith for unreal plugin (which doesn't support relocating the model in unreal space). There will be a fix eventually, and I will report is here, but if someone needs this they should ask and I will upload the 0.7.1 plugin somewhere for consumption
